Mwozniak

marcinwozniak0 opened the pull request.

marcinwozniak0 committed:
Add handling unexpected endings of the program

marcinwozniak0 committed:
Add cleaning of chat resources while detect user inactivity

marcinwozniak0 committed:
Review fixes from https://github.com/NMarcin/MessengerOnFileForLinux/pull/6. Added UserInactiveDetecorClass to replace old solution of user inactivity detecting

marcinwozniak0 commented on MessengerOnFileForLinux/ChatControl/src/UserInactivityDetector.cpp:
```diff
@@ -0,0 +1,29 @@
+#include "UserInactivityDetector.hpp"
```
ChatControl to według ciebie dobre miejsce na tą klase?

mnurzyns replied:
A jaki moduł odpowiada za sprzątanie po rozmowie? Może dać do do UserService? Nie wiem sam do końca

marcinwozniak0 replied:
Troche jest rozmyte to sprzątanie. Coś się sprząta w signal hnadlingu, coś w terminal funciotnality.Ja bym to zostawił tutaj. Pasuje pod ChatControl

mnurzyns commented on MessengerOnFileForLinux/ChatControl/src/ConversationControl.cpp:
```diff
@@ -107,17 +106,32 @@ bool ConversationControl::isMessagesToReadExist()
     }
 }
 
+void ConversationControl::handleInterlocutorInactivity() const
```
co robimy dokładnie ta funkcja, by widać tak jakbym wysyłała wiadomość. Czy nie powinna robić to przez MessageFramework?

marcinwozniak0 replied:
Nie wysyłamy wiadomości, tylko printujemy informacje na ekranie. Nie widzę potrzeby żeby taka informajca znajdowała sie w pliku z rozmową

mnurzyns replied:
Imho powinna znajdować się z pliku. Widzę to tak, profty z umieszczenia jej w pliku:
- dostajesz plik i widzisz już, że ktoś został wywalony
- ktoś kto skopiuje historię może pokazać drugiemu użytkownikowi, że go rozłączyło
- przyszłościowo pod ficzer z postmortemem, czyli jak Cię rozłączy to możesz wejść ponwnie do rozmowy by ją pobrać dla historii, jeśli nie zrobiłeś tego wcześniej, inaczej nikt by nie wiedział, że tak się skończyła rozmowa

mnurzyns reviewed (changes requested):
Brakuje mi ologowania SignalHandling

marcinwozniak0 committed:
Added doxygen to project. Doxygen documentation will be create when running cmake and generated file will be place in documentation/documentation.html.

marcinwozniak0 committed:
Added doxygen and graphviz instalation to setup_ci. It is necessary to run doxygen on CI platform

marcinwozniak0 committed:
Handling unexpected endings of the program after second review

Information about user inactivity is sending to chat file now. Added once_flag in handleInterlocutorInactivity to prevent multiple sending information about inactivity

marcinwozniak0 committed:
Merge branch 'master' of https://github.com/NMarcin/MessengerOnFileForLinux into mwozniak

marcinwozniak0 committed:
Merge branch 'master' into mwozniak

marcinwozniak0 committed:
Added automatic generation of doxygen documentation during executiong CI jobs

marcinwozniak0 committed:
Formatting change in .gitlab-ci to avoid CI compilation error

marcinwozniak0 committed:
Change in artifacts path in .gitlab-ci in MWOZNIAK_BUILD

marcinwozniak0 committed:
Revert "Change in artifacts path in .gitlab-ci in MWOZNIAK_BUILD"

This reverts commit cba4372b0928385bc6cc0157cbd588f75071be3e.

marcinwozniak0 committed:
Change filename of doxygen documentation generated file in CI script

marcinwozniak0 committed:
Use README.md as a doxygen main page
